app/policies/liquid_tag_policy.rb

```diff
@@ -0,0 +1,30 @@
+# Intentionally not inheriting from ApplicationPolicy because liquid tags behave
+# differently than the typical Model/Controller dynamic that Pundit assumes.
+class LiquidTagPolicy
+  attr_reader :user, :record
+
+  def initialize(user, record)
+    @user = user
+    @record = record
+  end
+
+  def initialize?
+    return true unless record.class.const_defined?("VALID_ROLES")
+    raise Pundit::NotAuthorizedError, "No user found" if record.class.const_defined?("VALID_ROLES") && user.nil?
```

Comment:
If we get to this line `VALID_ROLES` should be defined (since the previous line didn't return), so we don't need to check again for that, do we?

Reply:
Correct, good catch! This was my mistake from moving this check around and messing with Pundit. Updated 👍.